File: adventofcode.com/2024/01/2.py
import os
from collections import defaultdict

list_1 = []
list_2 = []

with open(os.path.dirname(__file__) + '/input.txt', mode='r') as f:
    for raw_line in f.readlines():
        values = raw_line.strip().split('   ')
        list_1.append(int(values[0]))
        list_2.append(int(values[1]))

list_2_counts = defaultdict(int)
for number in list_2:
    list_2_counts[number] += 1

score = 0
for number in list_1:
    score += number * list_2_counts[number]

print(score)  # 23046913

File: adventofcode.com/2024/01/2.rb
list_1 = []
list_2 = []

File.open(File.dirname(__FILE__) + "/input.txt", "r") do |f|
  f.each_line do |line|
    values = line.strip.split('   ')
    list_1.push(Integer(values[0]))
    list_2.push(Integer(values[1]))
  end
end

list_2_counts = Hash.new(0)
list_2.each do |number|
  list_2_counts[number] += 1
end

score = 0
list_1.each do |number|
  score += number * list_2_counts[number]
end

puts(score)  # 23046913

